data/data-pipeline/data_pipeline/content/config/generate_data_dictionary.py

```diff
@@ -0,0 +1,185 @@
+"""
+Quick script to generate a data dictionary for the download packet.
+
+A few notes:
+(1) This will ONLY crosswalk the downloadable CSV and the downloadable EXCEL file.
+    The SHP file has a LOT more columns, and we should almost certainly
+    make a data dictionary by hand once the column names are final.
+(2) I haven't given much thought to how the yaml for notes is stored and assume
+    this might change with the refactor
+"""
+import pathlib
+import yaml
+import pandas as pd
+import numpy as np
+from data_pipeline.config import settings
+from data_pipeline.score import field_names
+
+CEJST_SCORE_COLUMN_NAME = "score_name"
+CONFIG_PATH = settings.APP_ROOT / "content" / "config"
+
+EXCEL_CONFIG = CONFIG_PATH / "excel.yml"
+CSV_CONFIG = CONFIG_PATH / "csv.yml"
+SCORE_FIELD_CONFIG = CONFIG_PATH / "score_fields.yml"
+
+CSV_FORMAT = "csv_format"
+CSV_LABEL_FIELD = "csv_label"
+EXCEL_LABEL_FIELD = "excel_label"
+
+NOTES_FIELD = "notes"
+THRESHOLD_CATEGORY_FIELD = "threshold_category"
+CALCULATION_NOTES_FIELD = "calculation_notes"
+CSV_FIELD_TYPE_FIELD = "csv_field_type"
+
+COLUMNS_TO_KEEP = [
+    CEJST_SCORE_COLUMN_NAME,
+    CSV_FIELD_TYPE_FIELD,
+    CSV_LABEL_FIELD,
+    EXCEL_LABEL_FIELD,
+    CALCULATION_NOTES_FIELD,
+    THRESHOLD_CATEGORY_FIELD,
+    NOTES_FIELD,
+]
+
+# for re-typing & notes
+# note this doesn't perfectly square with field names
+PERCENTILE_STRING = field_names.PERCENTILE_FIELD_SUFFIX
+LOSS_RATE_STRING = "loss rate"
+LOW_STRING = "Low "
+ISLAND_STRING = "island areas"
+
+PERCENTILE_EXPLANATION = (
+    "All percentiles are floored (rounded down to the nearest percentile). "
+    + "For example, 89.7th percentile is rounded down to 89 for this field."
+)
+LOW_PERCENTILE_EXPLANATION = "This percentile is reversed, meaning the lowest raw numbers become the highest percentiles."
+ISLAND_AREAS_EXPLANATION = (
+    "Because not all data is available for the Nation, Puerto Rico, "
+    + "and the Island Territories, this uses different underlying data for the islands."
+)
+
+
+OUTPUT_PATH = (
+    settings.APP_ROOT / "data" / "score" / "downloadable" / "codebook.csv"
+)
+
+
+def _set_up_infrastructure(
+    yaml_path: pathlib.Path,
+    fields_to_store: list,
+):
+    """helper function to parse yaml"""
+    yaml_contents = yaml.safe_load(
```

Comment:
Oh, and it doesn't have to be in this PR, but at some point we should add some safety here by loading from YAMLs using a schema like Marshmallow to enforce that the YAML has the right format, acceptable values, etc. Probably part of the Big Refactor?

Reply:
Yup, this is this ticket #1327

Reply:
Not taking action here.